### DISCUSSION TITLE
How does `pnpm store prune` work?

### DISCUSSION THREAD

**Discussion by ChocolateLoverRaj** [pnpm/pnpm#4546] [Q&A] [answered]
I like the command `pnpm store prune`. I'm interested to know how it works. Does it scan every single `package.json` on the computer?

**Comment by zkochan** [comment, marked as answer]
No, it reads every file in the content-addressable store and removes those that have only 1 reference. 1 reference means that the file is not linked from any project on the computer.

https://github.com/pnpm/pnpm/blob/main/packages/package-store/src/storeController/prune.ts
